src/main/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterBase.java

```diff
@@ -0,0 +1,12 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import java.io.PrintWriter;
+
+public abstract class NamePrinterBase implements NamePrinter {
+    private String liTagPattern = "<li>%s</li>";
```

Comment:
please enter one line between class name and first variable

liTagPattern stand for ???

Reply:
liTagPattern is meaningless. Consider change it to more meaningful will help us understand it better.

Reply:
It's to be changed to `listItemTagPattern` instead.

Reply:
Looks good !!!